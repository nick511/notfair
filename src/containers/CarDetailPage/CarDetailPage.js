import React, { PureComponent } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import styles from './CarDetailPage.module.scss'
import * as actions from 'state/actions/carsActions'
import CarGallary from 'components/Car/CarGallary'
import CarDetail from 'components/Car/CarDetail'
import NotFound from 'components/NotFound/NotFound'

export class CarDetailPage extends PureComponent {
  componentDidMount = () => {
    this.props.actions.fetchCar(this.props.match.params.vin)
  }

  render() {
    const { data: car, failed } = this.props.car
    const actions = this.props.actions

    if (failed === true) return <NotFound />
    if (car == null) return null

    return (
      <div className={styles.component}>
        <CarGallary images={car.image_location_list} />
        <CarDetail car={car} favoriteCarAction={actions.favoriteCar} />
      </div>
    )
  }
}

const mapStateToProps = state => {
  return { car: state.car }
}

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(actions, dispatch),
})

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(CarDetailPage)
