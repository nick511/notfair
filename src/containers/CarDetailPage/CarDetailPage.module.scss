@import 'style/variables.scss';

.component {
  height: 100%;
  background-color: $background-color;
}
