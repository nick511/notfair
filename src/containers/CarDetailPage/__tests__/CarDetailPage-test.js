import React from 'react'
import { shallow } from 'enzyme'

import { CarDetailPage } from '../CarDetailPage'

describe('<CarDetailPage />', () => {
  let wrapper = null
  const match = {
    params: {
      vin: '19XFC2F59GE2276732016',
    },
  }
  const car = {
    id: '19XFC2F59GE2276732016',
    make: 'Honda',
    mileage: 35292,
    model: 'Civic Sedan',
    image_location_list: [],
  }
  const failed = false
  const actions = {
    fetchCar: jest.fn(),
    favoriteCar: jest.fn(),
  }

  beforeEach(() => {
    wrapper = shallow(
      <CarDetailPage
        match={match}
        car={car}
        failed={failed}
        actions={actions}
      />,
    )
  })

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
