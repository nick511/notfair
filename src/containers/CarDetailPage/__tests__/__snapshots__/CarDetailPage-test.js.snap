// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarDetailPage /> renders without crashing 1`] = `
<div
  className="component"
>
  <ErrorBoundary>
    <CarGallary
      images={Array []}
    />
  </ErrorBoundary>
  <ErrorBoundary>
    <CarDetail
      car={
        Object {
          "id": "19XFC2F59GE2276732016",
          "image_location_list": Array [],
          "make": "Honda",
          "mileage": 35292,
          "model": "Civic Sedan",
        }
      }
      favoriteCarAction={[MockFunction]}
    />
  </ErrorBoundary>
</div>
`;
